requiredRole.go

```diff
@@ -0,0 +1,27 @@
+// Copyright 2022 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cobra2snooty
+
+import (
+	"bytes"
+
+	"github.com/spf13/cobra"
+)
+
+func requiredRole(buf *bytes.Buffer, cmd *cobra.Command) {
+	if cmd.Annotations["requiredRole"] != "" {
+		buf.WriteString("\nTo use this command, the requesting user or API key must have the " + cmd.Annotations["requiredRole"] + " role.\n")
```

Comment:
Reading the rest of the long description, everything is framed in the second person. Should we do that here as well?

```suggestion
		buf.WriteString("\nTo use this command, you must have the " + cmd.Annotations["requiredRole"] + " role.\n")
```

Reply:
@melissamahoney-mongodb My concern with this change is that they can be authenticating either using their own user login (via atlas auth login) or with their API key. So there could conceivably be a scenario where their personal user login doesn't have the right permissions, but the API key does (or vice versa), and saying "you" doesn't capture that and might confuse people. What do you think of the following instead?
```suggestion
		buf.WriteString("\nTo use this command, you must authenticate with a user account or an API key that has the " + cmd.Annotations["requiredRole"] + " role.\n")
```
It's a little wordy, but I think it clears that up while using the second person.

Reply:
Yeah I like it!